climada/hazard/trop_cyclone.py

```diff
@@ -685,10 +685,8 @@ def compute_windfields(track, centroids, model, metric="equirect"):
     if np.count_nonzero(t_lat < 0) > np.count_nonzero(t_lat > 0):
         hemisphere = 'S'
     v_ang_rotate = [1.0, -1.0] if hemisphere == 'N' else [-1.0, 1.0]
-    v_ang_dir = np.array(v_ang_rotate)[..., :] * v_centr_normed[1:, :, ::-1]
-    v_ang = np.zeros_like(v_ang_dir)
-    v_ang[close_centr_msk[1:]] = v_ang_norm[close_centr_msk[1:], None] \
-                                 * v_ang_dir[close_centr_msk[1:]]
+    windfields = np.array(v_ang_rotate)[..., :] * v_centr_normed[:, :, ::-1]
+    windfields[close_centr_msk] *= v_ang_norm[close_centr_msk, None]
```

Comment:
Directly writing to the final `windfields` array avoids the temporary arrays `v_ang_dir`, `v_ang` and `v_full` (see below). That means we reduce the number of critical arrays by 6 (each of those arrays stores vectors, so has x and y component).